cloud/google/serviceaccount.go

```diff
@@ -26,68 +23,142 @@ import (
 )
 
 const (
-	ServiceAccountPrefix    = "k8s-machine-controller-"
-	ServiceAccount          = "service-account"
-	MachineControllerSecret = "machine-controller-credential"
+	MasterNodeServiceAccountPrefix        = "k8s-master"
+	WorkerNodeServiceAccountPrefix        = "k8s-worker"
+	MachineControllerServiceAccountPrefix = "k8s-machine-controller"
+	MachineControllerSecret               = "machine-controller-credential"
+	ClusterAnnotationPrefix               = "service-account-"
+)
+
+var (
+	MasterNodeRoles = []string{
+		"compute.instanceAdmin",
+		"compute.networkAdmin",
+		"compute.securityAdmin",
+		"compute.viewer",
+		"iam.serviceAccountUser",
+		"storage.admin",
+		"storage.objectViewer",
+	}
+	WorkerNodeRoles = []string{
```

Comment:
Why do nodes need these roles? (I'm uninformed)

Reply:
If possible, could this be a service account that the ingress controller has instead of every node it could run on? Like the machine controller does:

https://github.com/kubernetes-sigs/cluster-api/blob/master/cloud/google/serviceaccount.go#L75
https://github.com/kubernetes-sigs/cluster-api/blob/master/cloud/google/config/configtemplate.go#L174-L176
https://github.com/kubernetes-sigs/cluster-api/blob/master/cloud/google/config/configtemplate.go#L138-L139
https://github.com/kubernetes-sigs/cluster-api/blob/master/cloud/google/config/configtemplate.go#L144-L146